src/test/extensions/changelog.spec.ts

```diff
@@ -0,0 +1,120 @@
+import * as chai from "chai";
+import { expect } from "chai";
+chai.use(require("chai-as-promised"));
+import * as sinon from "sinon";
+
+import * as changelog from "../../extensions/changelog";
+import * as extensionApi from "../../extensions/extensionsApi";
+
+function testExtensionVersion(
+  version: string,
+  releaseNotes?: string
+): extensionApi.ExtensionVersion {
+  return {
+    name: `publishers/test/extensions/test/versions/${version}`,
+    ref: `test/test@${version}`,
+    hash: "abc123",
+    sourceDownloadUri: "https://google.com",
+    releaseNotes,
+    spec: {
+      name: "test",
+      version,
+      resources: [],
+      params: [],
+      sourceUrl: "https://google.com",
+    },
+  };
+}
+
+describe("changelog", () => {
+  describe("GetReleaseNotesForUpdate", () => {
+    let listExtensionVersionStub: sinon.SinonStub;
+
+    beforeEach(() => {
+      listExtensionVersionStub = sinon.stub(extensionApi, "listExtensionVersions");
+    });
+
+    afterEach(() => {
+      listExtensionVersionStub.restore();
+    });
+
+    it("should return release notes for each version in the update", async () => {
+      const extensionVersions: extensionApi.ExtensionVersion[] = [
+        testExtensionVersion("0.1.1", "foo"),
+        testExtensionVersion("0.1.2", "bar"),
```

Comment:
flip this to test the sort

Reply:
Oooh, good idea!

Reply:
Ahhh, on second though it won't actually cover that - the sorting happens in a different method that the one under test.